### PR TITLE
feat: add tolerance parameter to globalToLocal functions

### PR DESCRIPTION
**Pull request by paulgessinger**
As brought up on Mattermost and #493, the static tolerance in the `globalToLocal` functions might not be desirable.
This PR adds an additional argument that sets the tolerance. It is defaulted to `s_onSurfaceTolerance`, which was hardcoded before. This should be backwards compatible.

/cc @osbornjd

### CODE REVIEW

**Comment by asalzburger** [comment]
I like!

File: Core/src/Surfaces/CylinderSurface.cpp [hunk 1]
```diff
@@ -111,9 +111,13 @@ Acts::Vector3D Acts::CylinderSurface::localToGlobal(
...
     const GeometryContext& gctx, const Vector3D& position,
-    const Vector3D& /*unused*/) const {
-  // @todo check if s_onSurfaceTolerance would do here
-  double inttol = bounds().get(CylinderBounds::eR) * 0.0001;
+    const Vector3D& /*unused*/, double tolerance) const {
+  double inttol = tolerance;
+  if (tolerance == s_onSurfaceTolerance) {
+    // transform default value!
+    // @TODO: check if s_onSurfaceTolerance would do here
+    inttol = bounds().get(CylinderBounds::eR) * 0.0001;
```
Comment (asalzburger) [inline Core/src/Surfaces/CylinderSurface.cpp:119]: That actually comes from ATLAS, I think we try in a separate PR if we omit this awkward scaling.

**Comment by osbornjd** [comment]
This seems like a reasonable solution - then the tolerance level can be defined by the user (and therefore it is up to them to make sure they are right or not).

File: Core/src/Surfaces/CylinderSurface.cpp (hunk 1 above)
Comment (paulgessinger) [reply Core/src/Surfaces/CylinderSurface.cpp:119]: Yeah, this just replicates the current behavior if no explicit tolerance is given, without going back and questioning if this here make sense anymore.

**Review by asalzburger — APPROVED** [review]
Fine with me.

**Comment by XiaocongAi** [comment]
This is nice. It could be used to resolve #440.

**Comment by paulgessinger** [comment]
@asalzburger you might have to reapprove because I added a small additional change in the form of doc comments. That invalidated the previous review.

**Review by asalzburger — APPROVED** [review]

**Pull request merged**